drydock_provisioner/drivers/node/maasdriver/driver.py

```diff
@@ -926,6 +996,114 @@ def execute_task(self):
+                            status=hd_fields.TaskStatus.Complete,
+                            result=hd_fields.ActionResult.Failure,
+                            result_detail={'detail': 'Error accessing MaaS API', 'retry': True})
+                return
+
+            nodes = self.task.node_list
+
+            result_detail = {'detail': []}
+
+            worked = failed = False
+
+            for n in nodes:
+                try:
+                    self.logger.debug("Locating node %s for platform configuration" % (n))
+
+                    node = site_design.get_baremetal_node(n)
+                    machine = machine_list.identify_baremetal_node(node, update_name=False)
+
+                    if machine is None:
+                        self.logger.warning("Could not locate machine for node %s" % n)
+                        result_detail['detail'].append("Could not locate machine for node %s" % n)
+                        failed = True
+                        continue
+                except Exception as ex1:
+                    failed = True
+                    self.logger.error("Error locating machine for node %s: %s" % (n, str(ex1)))
+                    result_detail['detail'].append("Error locating machine for node %s" % (n))
+                    continue
+
+                try:    
+                    # Render the string of all kernel params for the node
+                    kp_string = ""
+
+                    for k,v in getattr(node, 'kernel_params', {}).items():
+                        if v == 'True':
+                            kp_string = kp_string + " %s" % (k)
+                        else:
+                            kp_string = kp_string + " %s=%s" % (k, v)
+
+                    if kp_string:
+                        # Check if the node has an existing kernel params tag
+                        node_kp_tag = tag_list.select("%s_kp" % (node.name))
+                        self.logger.info("Configuring kernel parameters for node %s" % (node.name))
+
+                        if node_kp_tag is None:
+                            self.logger.debug("Creating kernel_params tag for node %s: %s" % (node.name, kp_string))
+                            node_kp_tag = maas_tag.Tag(self.maas_client, name="%s_kp" % (node.name), kernel_opts=kp_string)
+                            node_kp_tag = tag_list.add(node_kp_tag)
+                            node_kp_tag.apply_to_node(machine.resource_id)
+                        else:
+                            self.logger.debug("Updating tag %s for node %s: %s" % (node_kp_tag.resource_id, node.name, kp_string))
+                            node_kp_tag.kernel_opts = kp_string
+                            node_kp_tag.update()
+
+                        self.logger.info("Applied kernel parameters to node %s" % n)
+                        result_detail['detail'].append("Applied kernel parameters to node %s" % (node.name))
+                        worked = True
```

Comment:
Worked is also set to True on line 1091. I would think that the flag should only be set to True once -- probably only on line 1091. These continue statements imply that the for loop continues executing across all nodes -- so the second problem I see is that worked and failed can both be True. However, line 1102 below doesn't include logic for setting PartialSuccess for this situation -- which you do in most other places.